src/Twig/GotenbergAssetRuntime.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Sensiolabs\GotenbergBundle\Twig;
+
+use Sensiolabs\GotenbergBundle\Builder\Pdf\AbstractChromiumPdfBuilder;
+use Sensiolabs\GotenbergBundle\Builder\Screenshot\AbstractChromiumScreenshotBuilder;
+
+/**
+ * @internal
+ */
+final class GotenbergAssetRuntime
+{
+    private AbstractChromiumPdfBuilder|AbstractChromiumScreenshotBuilder|null $builder = null;
+
+    public function setBuilder(AbstractChromiumPdfBuilder|AbstractChromiumScreenshotBuilder|null $builder): void
+    {
+        $this->builder = $builder;
+    }
+
+    /**
+     * This function is used to get the URL of an asset during the rendering
+     * of a PDF or a screenshot with the Gotenberg client.
+     *
+     * It only works if the builder is an instance of AbstractChromiumPdfBuilder
+     * or AbstractChromiumScreenshotBuilder.
+     */
+    public function getAssetUrl(string $path): string
+    {
+        if (null === $this->builder) {
+            throw new \LogicException('You need to extend from AbstractChromiumPdfBuilder to use "gotenberg_asset" function.');
```

Comment:
```suggestion
            throw new \LogicException('The gotenberg_asset function must be used in a Gotenberg context. You can use it via the AbstractChromiumPdfBuilder or AbstractChromiumScreenshotBuilder class.');
```
Should be easier to understand.

Reply:
I took the original one, but yeah this looks a bit weird. 

To be honest i would not specify class names here at all.. It's a bit weird no ?  

And add a dedicated interface/VO like "Gotenberg\Assets" (or "AssetSet" or "TemplateAssetBag" or ...)  and just use $assets->add(...).  

Could even use a default one if none is set.

In the end, I think this function could very much work for everyone... the AssetBag would simply **not** be collected  after the render of the template. 

WDYT ?

Reply:
> To be honest i would not specify class names here at all.. It's a bit weird no ?

Indeed it is not very helpful.

> And add a dedicated interface/VO like "Gotenberg\Assets" (or "AssetSet" or "TemplateAssetBag" or ...) and just use $assets->add(...).

I hope I can have some time to rework builders in that way. I started to work with a HeadersBag and a BodyBag, but an AssetBag should be added too.

Reply:
AssetBag->addAsset($asset)

Here, $asset can be:
* A logical path: mapped by assetmapper/webpack -> Package::getUrl.
* An absolute URL: directly accepted.
* A local file: either streamed or converted into a signed URL.

This approach should simplify things quite a bit.

And then a second twig function to inline _**some**_ assets (base64 images for instance, work good for header/footer things)